### ISSUE TITLE
[question]When vscode remote SHH to host, how can I send the simulation back to local

### ISSUE THREAD

**Issue by huang12zheng** [Dart-Code/Dart-Code#2136]
<img width="1055" alt="屏幕快照 2019-11-29 上午10 06 30" src="https://user-images.githubusercontent.com/28038074/69839080-0c0da480-1291-11ea-8bca-e036c887eada.png">

* any suggestion?
---
my idea
+ condition:
android studio/xcode in My Locally , 
> flutter devices find My Locally devices , It would run,and show simulation.

+ condition
android studio in host
> I don't have any idea

**Comment by huang12zheng** [comment]
I find a issue in stackoverflow -_-
https://stackoverflow.com/questions/58441350/developing-flutter-with-vscode-and-wsl2?r=SearchResults

**Comment by huang12zheng** [comment]
Reopen in the hope of a more convenient solution

**Comment by huang12zheng** [comment]
```
# MAC
adb tcpip 5555
```
```
# Linux 
adb connect IP: 5555
```
failed to connect to '192.168.50.243:5555': Connection refused

**Comment by DanTup** [comment]
I think your last comment should work. Are you using the IP address of the phone, and not the Mac?

Can you ping the phone's IP address from the terminal inside VS Code?

(I presume you've enabled dev mode etc. on the phone, and it works when launching Flutter apps directly from the Mac?)

**Comment by DanTup** [comment]
(Note: if you're using a simulator instead of a physical phone, then I guess this needs to be the IP of the simulator - though I'm not sure how best to get that)

**Comment by huang12zheng** [comment]
vscode terminal (linux) 192.168.50.149
phone ip 192.168.50.127
mac 192.168.50.243

~~if I `adb connect` to phone,it show` failed to authenticate to 192.168.50.127:5555 `~~
------
it connect ,
![屏幕快照 2019-11-29 下午9 10 15](https://user-images.githubusercontent.com/28038074/69871024-b0710480-12ec-11ea-80d3-9d2692109884.png)

if I connect to 192.168.50.127:5557

**Comment by huang12zheng** [comment]
but it is Real machine
emulator is not there yet.
Just Thank you very much for `Are you using the IP address of the phone`

**Comment by huang12zheng** [comment]
I would try replacing vscode terminal (linux) 192.168.50.149 to Cloud
If it is work, I think is excited to me

**Comment by huang12zheng** [comment]
close, and would open in future.

**Comment by DanTup** [comment]
@huang12zheng I'm not sure from your response - did this work? Is everything resolved?

**Comment by huang12zheng** [comment]
it could work on real manchine.  But emulator is not ok. 
> I think it is difficult to solve.

**Comment by huang12zheng** [comment]
now, I want to start a emulator in web. but there is still some problem to solve.
https://github.com/google/android-emulator-container-scripts/issues/78

**Comment by DanTup** [comment]
> it could work on real manchine. But emulator is not ok.

Does this mean you could not get it to work? I would expect it to work, even with an Android Emulator. If not, I can do some testing.

**Comment by DanTup** [comment]
I did some testing of running the emulator on the host, but I was unable to connect Flutter inside a Docker container back out to an emulator running on the host. There seems to be all sorts of restrictions on both the networking between the host and the emulator, and between the host and the Docker container. I tried running the `adb daemon` with `adb -a nodaemon server start` and using adb forward` to forward the port, but it still didn't seem to work - I couldn't get a connection from inside the container back through the host to the emulator.

There may be a way to make this work, but I hit a wall (and tbh if you're setting up the Android SDK on the host, there's probably not much reason to push Flutter into the container).
